redash/query_runner/pg.py

```diff
@@ -260,6 +265,12 @@ def _get_tables(self, schema):
         )
         SELECT table_name, table_schema, column_name
         FROM tables
+        WHERE
+            HAS_SCHEMA_PRIVILEGE(table_schema, 'USAGE') AND
+            (
+                table_schema IN (SELECT schemaname FROM SVV_EXTERNAL_SCHEMAS) OR
+                HAS_TABLE_PRIVILEGE(table_schema || '.' || table_name, 'SELECT')
```

Comment:
Just to make sure I understand: when a table is an external one, then `HAS_TABLE_PRIVILEGE` does not apply and having USAGE permission for the schema is enough?

Reply:
Never mind -- just read your superb description that explains all of that.